### DISCUSSION TITLE
Add optional Description to Dashboard Annotations

### DISCUSSION THREAD

**Discussion by Mario-Hofstaetter** [grafana/grafana#38243] [Dashboards, Panels & Visualizations]
**What would you like to be added**:

Lately it is possible to add an optional `Description` to dashboard variables, which is shown in the tooltip (see below).

**It would be nice to be able to add a Description to `Annotations`** too, further explaining to the user what particular Annotation Query toggle might show to him.

If the toogle is visible, it should use the description as tooltip same as variables do, if the toggle is `Hidden`, the Description is _documenting_ the meaning of the annotation query in the ui.

![grafik](https://user-images.githubusercontent.com/33002073/112556157-df589700-8dc9-11eb-853d-1ff484070131.png)

VS

![grafik](https://user-images.githubusercontent.com/33002073/112555881-55a8c980-8dc9-11eb-8363-d335e1b5aefb.png)

**Why is this needed**:

When using multiple Annotation Queries on a dashboard, their name ought be brief so the toolbar is not spammed. But short names do not always explain to the user what the annotations are exactly.

Example running `Grafana v7.5.0 (b144fb1a9f)` at <https://play.grafana.org/d/000000010>:

Variable Tooltip vs no tooltip on multiple annotation toggles:
![n9Tsia0xop](https://user-images.githubusercontent.com/33002073/112556002-9b659200-8dc9-11eb-9f6d-0bfaee2f6af0.gif)

Using a more verbose name for annotations takes a lot of screen real estate, especially if multiple dashboard variables are needed too:

![grafik](https://user-images.githubusercontent.com/33002073/112556533-b1278700-8dca-11eb-8128-9758b9848213.png)

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
